tools/configuration-converter.sh

```diff
@@ -0,0 +1,140 @@
+    echo "  <input>         : Input file (required)"
+    echo "  [output]        : Output file (optional, defaults to input's base directory with '-sail.yaml' suffix)"
+    echo "  -n <namespace>  : Namespace (optional, defaults to 'istio-system')"
+    echo "  -v <version>    : Istio Version (optional)"
+    exit 1
+}
+
+# Ensure at least one file argument is provided
+if [[ $# -lt 1 ]]; then
+    usage
+fi
+
+# Initialize variables
+INPUT=""
+OUTPUT=""
+NAMESPACE="istio-system"
+VERSION=""
+
+# Identify the first and second file paths (treat the first file as INPUT)
+for arg in "$@"; do
+    if [[ -f "$arg" && -z "$INPUT" ]]; then
+        INPUT="$arg"  # First file is the INPUT
+    elif [[ -f "$arg" && -z "$OUTPUT" ]]; then
+        OUTPUT="$arg"  # Second file is the OUTPUT
+    fi
+done
+
+# Ensure the input file is provided and is valid
+if [[ -z "$INPUT" || ! -f "$INPUT" ]]; then
+    echo "Error: Input file is missing or invalid."
+    usage
+fi
+
+# If OUTPUT is not specified, generate a default output file name
+if [[ -z "$OUTPUT" ]]; then
+    OUTPUT="$(dirname "$INPUT")/$(basename "$INPUT" .yaml)-sail.yaml"
+fi
+
+# Ensure the output is not a directory
+if [[ -d "$OUTPUT" ]]; then
+    echo "Error: OUTPUT must be a file, not a directory."
+    exit 1
+fi
+
+# Parse optional namespace (-n) and version (-v) arguments
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        -n)
+            NAMESPACE="$2"
+            shift 2
+            ;;
+        -v)
+            VERSION="$2"
+            shift 2
+            ;;
+        *)
+            shift
+            ;;
+    esac
+done
+
+
+if ! command -v yq &>/dev/null; then
+    echo "Error: 'yq' is not installed. Please install it before running the script."
+    exit 1
+fi
+
+function add_mandatory_fields(){
+    # Base yq command without the version field
+    yq_cmd=".apiVersion = \"sailoperator.io/v1\" 
+            | .kind = \"Istio\" 
+            | (select(.spec.meshConfig) | .spec.values.meshConfig) = .spec.meshConfig 
+            | (select(.spec.values.istio_cni) | .spec.values.pilot.cni) = .spec.values.istio_cni 
+            | .metadata.name = \"default\" 
+            | .spec.namespace = \"$NAMESPACE\" 
+            | del(.spec.values.istio_cni)
+            | del(.spec.meshConfig)
+            | del(.spec.hub) 
+            | del(.spec.tag) 
+            | del(.spec.values.gateways)"
+    
+    # If VERSION is not empty, add .spec.version
+    if [[ -n "$VERSION" ]]; then
+        yq_cmd="$yq_cmd | .spec.version = \"$VERSION\""
+    fi
+
+    # Execute the yq command with the constructed string
+    yq -i eval "$yq_cmd" "$OUTPUT"    
+}
+
+
+#Convert boolean values to string if they are under *.env
+function boolean_2_string(){
+    yq -i -e '(.spec.values.[].env.[] | select(. == true)) |= "true" 
+    | (.spec.values.[].env.[] | select(. == false)) |= "false"' "$OUTPUT"
+    yq -i -e 'del(.. | select(tag == "!!seq" and length == 0))' "$OUTPUT"
```

Comment:
Still two separate commands? :)

Maybe I missed your comment explaining why it can't be a single command.